src/utils/constants.ts

```diff
@@ -9,23 +13,122 @@ export const ERC20_TRANSFER_SIGNATURE = '0xa9059cbb';
 export const ERC20_APPROVE_SIGNATURE = '0x095ea7b3';
 export const DEFAULT_TOKEN_DECIMALS = 18;
 
-export const NETWORK_NAMES = {
-  '1': 'mainnet',
-  '4': 'rinkeby',
-  '100': 'xdai',
-  '42161': 'arbitrum',
-  '421611': 'arbitrumTestnet',
-  '1337': 'localhost',
-};
-
 const defaultInfuraKey = process.env.REACT_APP_KEY_INFURA_API_KEY;
-export const DEFAULT_RPC_URLS = {
-  '1': `https://mainnet.infura.io/v3/${defaultInfuraKey}`,
-  '4': `https://rinkeby.infura.io/v3/${defaultInfuraKey}`,
-  '100': 'https://rpc.xdaichain.com/',
-  '42161': `https://arbitrum-mainnet.infura.io/v3/${defaultInfuraKey}`,
-  '421611': `https://arbitrum-rinkeby.infura.io/v3/${defaultInfuraKey}`,
-};
+
+export const NETWORKS: {
+  id: number,
+  name: string,
+  displayName: string,
+  defaultRpc: string,
+  nativeAsset: {
+    name: string,
+    symbol: string,
+    decimals: number
+  },
+  blockExplorer?: string,
+  icon?: string
+}[] = [
+  {
+    id: 1,
+    name: 'mainnet',
+    displayName: 'Ethereum Mainnet',
+    defaultRpc: `https://mainnet.infura.io/v3/${defaultInfuraKey}`,
+    nativeAsset: {
+      name: "Ethereum",
+      symbol: 'ETH',
+      decimals: 18
+    },
+    blockExplorer: 'https://etherscan.io',
+    icon: ethereumIcon
+  },
+  {
+    id: 4,
+    name: 'rinkeby',
+    displayName: 'Rinkeby Testnet',
+    defaultRpc: `https://rinkeby.infura.io/v3/${defaultInfuraKey}`,
+    nativeAsset: {
+      name: "Ethereum",
+      symbol: 'ETH',
+      decimals: 18
+    },
+    blockExplorer: 'https://rinkeby.etherscan.io/',
+    icon: ethereumIcon
+  },
+  {
+    id: 100,
+    name: 'xdai',
+    displayName: 'xDai Chain',
+    defaultRpc: `https://rpc.xdaichain.com/`,
+    nativeAsset: {
+      name: "xDai",
+      symbol: 'xDAI',
+      decimals: 18
+    },
+    blockExplorer: 'https://blockscout.com/xdai/mainnet/',
+    icon: xdaiIcon
+  },
+  {
+    id: 42161,
+    name: 'arbitrum',
+    displayName: 'Arbitrum One',
+    defaultRpc: `https://arbitrum-mainnet.infura.io/v3/${defaultInfuraKey}`,
+    nativeAsset: {
+      name: "Ethereum",
+      symbol: 'ETH',
+      decimals: 18
+    },
+    blockExplorer: 'https://arbiscan.io/',
+    icon: arbitrumIcon
+  },
+  {
+    id: 421611,
+    name: 'arbitrumTestnet',
+    displayName: 'Arbitrum Testnet',
+    defaultRpc: `https://arbitrum-rinkeby.infura.io/v3/${defaultInfuraKey}`,
+    nativeAsset: {
+      name: "Ethereum",
+      symbol: 'ETH',
+      decimals: 18
+    },
+    blockExplorer: 'https://rinkeby-explorer.arbitrum.io/',
+    icon: arbitrumIcon
+  },
+  {
+    id: 1337,
+    name: 'localhost',
+    displayName: 'Localhost',
+    defaultRpc: `http://localhost:8545`,
+    nativeAsset: {
+      name: "Ethereum",
+      symbol: 'ETH',
+      decimals: 18
+    },
+  },
+];
+
+export const NETWORK_NAMES: { [chainId: string]: string } = NETWORKS.reduce(

```

Comment:
and each of this functions get a subtype of the types deifned before?

Reply:
That's very elegant!

Reply:
yeah thats a way! another way I have seen is: 
type ChainID { 
   id: string
}
   
type Chain  = ChainID & {